pkg/objstore/azure/helpers.go

```diff
@@ -38,20 +38,31 @@ func getAzureStorageCredentials(conf Config) (blob.Credential, error) {
 	if conf.MSIResource != "" {
 		msiConfig := auth.NewMSIConfig()
 		msiConfig.Resource = conf.MSIResource
-
 		azureServicePrincipalToken, err := msiConfig.ServicePrincipalToken()
 		if err != nil {
 			return nil, err
 		}
 
-		// Get a new token.
+		var tokenRefresher blob.TokenRefresher = func(credential blob.TokenCredential) time.Duration {
```

Comment:
what about defining this function outside? (normally) ;p

Reply:
Cant really do that as it need access to the spt.